wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -0,0 +1,1777 @@
+			return;
+		}
+
+		$table_name = $this->unquote_sqlite_identifier(
+			$this->translate( $node->get_descendant_node( 'tableName' ) )
+		);
+
+		// Save information to information schema tables.
+		$this->information_schema_builder->record_create_table( $node );
+
+		// Generate CREATE TABLE statement from the information schema tables.
+		$queries            = $this->get_sqlite_create_table_statement( $table_name );
+		$create_table_query = $queries[0];
+		$constraint_queries = array_slice( $queries, 1 );
+
+		$this->execute_sqlite_query( $create_table_query );
+		$this->results      = $this->last_exec_returned;
+		$this->return_value = $this->results;
+
+		foreach ( $constraint_queries as $query ) {
+			$this->execute_sqlite_query( $query );
+		}
+	}
+
+	private function execute_alter_table_statement( WP_Parser_Node $node ): void {
+		$table_name = $this->unquote_sqlite_identifier(
+			$this->translate( $node->get_descendant_node( 'tableRef' ) )
+		);
+
+		// Save all column names from the original table.
+		$column_names = $this->execute_sqlite_query(
+			'SELECT COLUMN_NAME FROM _mysql_information_schema_columns WHERE table_schema = ? AND table_name = ?',
+			array( $this->db_name, $table_name )
+		)->fetchAll( PDO::FETCH_COLUMN );
+
+		// Track column renames and removals.
+		$column_map = array_combine( $column_names, $column_names );
+		foreach ( $node->get_descendant_nodes( 'alterListItem' ) as $action ) {
+			$first_token = $action->get_child_token();
+
+			if ( WP_MySQL_Lexer::DROP_SYMBOL === $first_token->id ) {
+				$name = $this->translate( $action->get_child_node( 'columnInternalRef' ) );
+				if ( null !== $name ) {
+					$name = $this->unquote_sqlite_identifier( $name );
+					unset( $column_map[ $name ] );
+				}
+			}
+
+			if ( WP_MySQL_Lexer::CHANGE_SYMBOL === $first_token->id ) {
+				$old_name = $this->unquote_sqlite_identifier(
+					$this->translate( $action->get_child_node( 'columnInternalRef' ) )
+				);
+				$new_name = $this->unquote_sqlite_identifier(
+					$this->translate( $action->get_child_node( 'identifier' ) )
+				);
+
+				$column_map[ $old_name ] = $new_name;
+			}
+
+			if ( WP_MySQL_Lexer::RENAME_SYMBOL === $first_token->id ) {
+				$column_ref = $action->get_child_node( 'columnInternalRef' );
+				if ( null !== $column_ref ) {
+					$old_name = $this->unquote_sqlite_identifier(
+						$this->translate( $column_ref )
+					);
+					$new_name = $this->unquote_sqlite_identifier(
+						$this->translate( $action->get_child_node( 'identifier' ) )
+					);
+
+					$column_map[ $old_name ] = $new_name;
+				}
+			}
+		}
+
+		$this->information_schema_builder->record_alter_table( $node );
+
+		/*
+		 * See:
+		 *   https://www.sqlite.org/lang_altertable.html#making_other_kinds_of_table_schema_changes
+		 */
+
+		// 1. If foreign key constraints are enabled, disable them.
+		// @TODO
+
+		// 2. Create a new table with the new schema.
+		$tmp_table_name     = "_tmp__{$table_name}_" . uniqid();
```

Comment:
nitpick: we'll need a reserved table prefix for any system operations, and we'll want to make sure explicit user queries won't affect any tables with that prefix